data/unaligned_masked_dataset.py

```diff
@@ -0,0 +1,152 @@
+import os.path
+import random
+
+import numpy as np
+import torchvision.transforms as transforms
+from PIL import Image
+
+from data.base_dataset import BaseDataset
+from data.image_folder import make_dataset
+
+
+class TransformWithMask:
+    def __init__(self, opt, grayscale, method=Image.BICUBIC):
+        self.grayscale = transforms.Grayscale(1) if grayscale else None
+        self.resize = transforms.Resize((opt.load_size, opt.load_size), method)
+        self.to_tensor = transforms.ToTensor()
+        self.normalize = (
+            transforms.Normalize((0.5,), (0.5,))
+            if grayscale
+            else transforms.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5))
+        )
+        self.crop_size = opt.crop_size
+        self.load_size = opt.load_size
+
+    def random_crop(self, img, mask):
+        x = random.randint(0, np.maximum(0, self.load_size - self.crop_size))
+        y = random.randint(0, np.maximum(0, self.load_size - self.crop_size))
+        cropped_img = img[:, x : x + self.crop_size, y : y + self.crop_size]
+        cropped_mask = mask[:, x : x + self.crop_size, y : y + self.crop_size]
+        return cropped_img, cropped_mask
+
+    def __call__(self, img, mask):
+        if self.grayscale is not None:
+            img = self.grayscale(img)
+            mask = self.grayscale(mask)
+        img = self.to_tensor(self.resize(img))
+        img = self.normalize(img)
+        mask = self.to_tensor(self.resize(mask))
+        # convert float to binary
+        mask = (mask > 0.5).float()
```

Comment:
Don't worry. Your code is great.
I think this line should be `mask = (mask >= 0.5).float()` kana.